Juniper upgrade

felipemontoya opened the pull request:
This PR makes the plugin work for juniper instances

**Changes and justification**

* [Overriding](https://github.com/eduNEXT/eox-theming/pull/27/commits/f3c89b23b1432aa9a06b5659186d1d7ba4abb22e) of get_theme_template_sources() method
The get_dirs() method from the EoxThemeFilesystemLoader class is not returning the template directories of the current theme, it only returns a list with the parent/grandparent template dirs and the engine dirs (the engine directories list only has the
paths specified in the [common](https://github.com/eduNEXT/edunext-platform/blob/master/lms/envs/common.py#L620-L624) settings file). As a consequence, the Django templates are being loaded from the parent/grandparent theme and not from the main theme.
To solve this problem, get_theme_template_sources() was overridden so that it returns not only a list with the current theme template dirs, but also the parent and grandparent theme template dirs (in order), and since this method is called during the [init](https://github.com/eduNEXT/edunext-platform/blob/master/openedx/core/djangoapps/theming/template_loaders.py#L36) method of the ThemeFilesystemLoader class, the template sources of all of the themes are part of the dirs attribute of the object, so when the get_dirs() method gets called, it returns this list of dirs. 

* Deletion of the get_dirs() method
Since all of the themes are now part of the ThemeFilesystemLoader dirs, is not necessary to override the get_dirs() method to include the template directories of the themes. This method was getting called multiple times during the requests, and on each call, it had to find the template dirs for the parent and grandparent theme,  now it only has to return the dirs of the ThemeFilesystemLoader object.

* [Modification](https://github.com/eduNEXT/eox-theming/pull/27/commits/a49efd147fe52e111507ed7f85bd3ba0971c4316#) of the adjust_uri  method  
Each time the calling template (calling_uri) is located inside a theme but is not trying to inherit itself  
**get_template_path_with_theme(strip_site_theme_templates_path(relative_uri))** would get called twice. For example _calling_uri: 'test-3/lms/templates/header/navbar-logo-header.html'_  and _uri: '/static_content.html'_ .
To avoid this, the var adjusted_uri  was added to keep the result of the function call so the comparison and return of the value can be made with only one call.

* Overriding of get_themed_packages() method to avoid including non themed assets during the collect static command,
so the themes assets are loaded in the correct order for the sites.

* Modification of the settings in the settings/test.py file to disable eox-theming for the platform tests

**Additional comments**

* The method [get_template](https://github.com/eduNEXT/eox-theming/blob/master/eox_theming/theming/paths.py#L50) calls the super [method](https://github.com/eduNEXT/edx-platform/blob/master/common/djangoapps/edxmako/paths.py#L86) from the  DynamicTemplateLookup class, but I do not think this is necessary since the overridden method is already responsible for locating a template in the site themes or fallback to the default template directories (in case the template could not be located). I was thinking maybe we could make this method call the super method from the base class TemplateLookup instead. Perhaps this is not something that would make a huge difference,  but I would like to know what you guys think.

morenol commented on eox_theming/theming/storage.py:
```diff
@@ -92,9 +92,7 @@ def url(self, name):
-        PipelineForgivingStorage,
-        OptimizedFilesMixin,
-        ThemeCachedFilesMixin,
+        ProductionStorageMixin,
```
This change also includes the ThemePipelineMixin, if I recall fine, that had problems with the propagation of static assets from the parent themes. I might be wrong but I think that it is something that needs to be checked running make lms-static

morenol replied:
Are you having problems with static files? Did you check this?

andrey-canon replied:
@morenol yes, right now some course images are not loading, any suggestion to fix it ?

morenol commented on eox_theming/edxapp_wrapper/storage.py:
```diff
@@ -24,6 +24,15 @@ def get_themepipeline_mixin():
-    """Return the Theme pipeline mixin when called during runtime"""
+    """
+    Return the PipelineForgiving storage when called during runtime
+    No longer used on juniper.
```
If we are not going to use this in Juniper, does it make sense to keep this function?

felipemontoya replied:
Yes because this is the storage interface. This needs to remain if we need to use it in ironwood for older installs.
We could remove it once we drop support for ironwood

andrey-canon replied:
But I think this code already drops the support for ironwood, the following code will try to import something that doesn't exist in ironwood and if we change the backend setting to  `'eox_theming.edxapp_wrapper.backends.i_storage'` the method `get_production_mixin` doesn't exist 
![image](https://user-images.githubusercontent.com/36200299/98391915-49dc8680-2025-11eb-8d2d-030b78dd2e2b.png)

imo if we want to keep the compatibility  a method get_production_mixin  should be created in the `'eox_theming.edxapp_wrapper.backends.i_storage'` which returns a mixin with the classes which were removed 
![image](https://user-images.githubusercontent.com/36200299/98392814-71801e80-2026-11eb-9f9b-975e9d705f92.png)

felipemontoya commented:
I just squashed all the changes, removed the dev only settings and force-pushed.

This are the changes that were used in the dev environment to fix the themes. They were approved by @amalbas for that task.

andrey-canon commented on CHANGELOG.rst:
```diff
@@ -0,0 +1,32 @@
+==========
+
+..
+   All enhancements and patches to eox-core will be documented
```
eox-core ???

andrey-canon commented on eox_theming/edxapp_wrapper/backends/i_storage.py:
```diff
@@ -27,3 +28,22 @@ def get_themepipeline_mixin():
+    Combines several mixins that provide additional functionality.
+    We use this version on production.
+    """
+    pass
```
useless-pass
